Upstream-Status: Inappropriate [upstream is dead]

Signed-off-by: Ricardo Ribalda <ricardo@ribalda.com>

diff --git a/native/fdlibm/ieeefp.h b/native/fdlibm/ieeefp.h
index c230bbb..700dd3a 100644
--- a/native/fdlibm/ieeefp.h
+++ b/native/fdlibm/ieeefp.h
@@ -158,6 +158,10 @@
 #define __IEEE_BIG_ENDIAN
 #endif
 
+#ifdef __aarch64__
+#define __IEEE_LITTLE_ENDIAN
+#endif
+
 #ifndef __IEEE_BIG_ENDIAN
 #ifndef __IEEE_LITTLE_ENDIAN
 #error Endianess not declared!!
